test/integration/build/test_build.py

```diff
@@ -81,6 +81,63 @@ def test_app_without_rockspec(cartridge_cmd, project_without_dependencies):
     assert 'Application directory should contain rockspec' in output
 
 
+def test_building_with_rockspec_specifying_bad_filename(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    rockspec_name = "{}-scm-1.rockspec".format(project.name)
+    rocks_make_output = "Error: File not found: {}".format(rockspec_name)
+
+    os.remove(project.rockspec_path)
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        rockspec_name,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 1, 'Building project should fail'
+    assert rocks_make_output in output
+
+
+def test_building_with_rockspec_specifying(cartridge_cmd, project_without_dependencies):
+    project = project_without_dependencies
+
+    prebuild_output = "pre-build hook output"
+    rockspec_name = "{}-scm-1.rockspec".format(project.name)
+
+    with open(os.path.join(project.path, 'cartridge.pre-build'), 'w') as f:
+        prebuild_script_lines = [
+            "#!/bin/sh",
+            "echo \"{}\"".format(prebuild_output)
+        ]
+        f.write('\n'.join(prebuild_script_lines))
+
+    build_logs = [
+        'Build application in',
+        'Running `cartridge.pre-build`',
+        'Running `tarantoolctl rocks make {}`'.format(rockspec_name),
+        'Application was successfully built',
+    ]
+
+    cmd = [
+        cartridge_cmd,
+        "build",
+        "--spec",
+        rockspec_name,
+    ]
+
+    rc, output = run_command_and_get_output(cmd, cwd=project.path)
+    assert rc == 0
+    assert all([log in output for log in build_logs])
+    assert prebuild_output not in output
+
+    # check that all expected rocks was installed
+    files = recursive_listdir(project.path)
+    assert '.rocks' in files
+    assert all([rock in files for rock in project.rocks_content])
+
+
```

Comment:
Think about this test cases:

* User has two rockspecs in project root: What if he didn't pass `--spec` flag? what if he passed one of existing specs? what if he passed bad path?
* User hasn't rockspecs in project root:  What if he didn't pass `--spec` flag? What if he passed path to a spec from other directory (I'm not sure that it's supported by luarocks, please, check it).
* User has one scm-1 rockspec (default case)
* User has one scm-2 rockspec: How does luarocks handle this?